app/src/main/java/ch/hikemate/app/utils/MapUtils.kt

```diff
@@ -2,108 +2,134 @@ package ch.hikemate.app.utils
- *   calculate the center.
- * @return A `GeoPoint` representing the geographical center of the provided bounds.
- */
-fun getGeographicalCenter(bounds: Bounds): GeoPoint {
-  Log.d(
-      "Bounds",
-      "minLat: ${bounds.minLat}, maxLat: ${bounds.maxLat}, minLon: ${bounds.minLon}, maxLon: ${bounds.maxLon}")
-  val minLong = bounds.minLon
-  val maxLong = bounds.maxLon
-  val minLat = bounds.minLat
-  val maxLat = bounds.maxLat
-
-  val centerLat = (minLat + maxLat) / 2
-  var centerLong = (minLong + maxLong) / 2
-
-  // Adjust if the longitude crosses the Date Line (i.e., difference > 180 degrees)
-  if (maxLong - minLong > 180) {
-    centerLong = (minLong + maxLong + 360) / 2
-  } else if (minLong - maxLong > 180) {
-    centerLong = (minLong + maxLong - 360) / 2
+object MapUtils {
+
+  /**
+   * Shows a hike on the map.
+   *
+   * @param mapView The map view where the hike will be shown.
+   * @param hike The hike to be shown.
+   * @param color The color of the hike.
+   */
+  fun showHikeOnMap(
+      mapView: MapView,
+      hike: HikeRoute,
+      color: Int,
+      onLineClick: () -> Unit,
+  ) {
+    val line = Polyline()
+
+    line.setPoints(hike.ways.map { GeoPoint(it.lat, it.lon) })
+    line.outlinePaint.color = color
+    line.outlinePaint.strokeWidth = MapScreen.STROKE_WIDTH
+
+    line.setOnClickListener { _, _, _ ->
+      onLineClick()
+      true
+    }
+
+    mapView.overlays.add(line)
   }
 
-  // Normalize longitude to be in the range -180 to 180
-  if (centerLong > 180) {
-    centerLong -= 360
-  } else if (centerLong < -180) {
-    centerLong += 360
+  /**
+   * Calculates the geographical center of a given bounding box. This function computes the center
+   * latitude as the average of the minimum and maximum latitudes, and the center longitude as the
+   * average of the minimum and maximum longitudes. It also handles edge cases where the longitude
+   * crosses the Date Line (±180 degrees), adjusting the center longitude accordingly to ensure it
+   * stays within the valid range of -180 to 180 degrees.
+   *
+   * @param bounds The bounding box (minLat, maxLat, minLon, maxLon) defining the area for which to
+   *   calculate the center.
+   * @return A `GeoPoint` representing the geographical center of the provided bounds.
+   */
+  fun getGeographicalCenter(bounds: Bounds): GeoPoint {
+    Log.d(
+        "MapUtils",
+        "minLat: ${bounds.minLat}, maxLat: ${bounds.maxLat}, minLon: ${bounds.minLon}, maxLon: ${bounds.maxLon}")
+    val minLong = bounds.minLon
+    val maxLong = bounds.maxLon
+    val minLat = bounds.minLat
+    val maxLat = bounds.maxLat
+
+    val centerLat = (minLat + maxLat) / 2
+    var centerLong = (minLong + maxLong) / 2
+
+    // Adjust if the longitude crosses the Date Line (i.e., difference > 180 degrees)
+    if (maxLong - minLong > 180) {
+      centerLong = (minLong + maxLong + 360) / 2
+    }
```

Comment:
Suggestion: if I'm not mistaken, this check does not guarantee that `centerLong` is within valid bounds (could be greater than 180).

Would this work?

```suggestion
    // Adjust if the longitude crosses the Date Line (i.e., difference > 180 degrees)
    if (maxLong - minLong > 180) {
      centerLong = (minLong + maxLong + 360) % 360 - 180
    }
```

Or should we rather do this?

```suggestion
    // Adjust if the longitude crosses the Date Line (i.e., difference > 180 degrees)
    if (maxLong - minLong > 180) {
      centerLong = ((minLong + maxLong + 360) / 2) % 360
    }
```

Reply:
I'm pretty sure it works. The +360 "normalises" the negative longitude, essentially unwrapping it and moving it into the "positive" side of the map, allowing us to add the two distances as if they were in a straight line, without the dateline in between them. I've changed the order of the terms to make it more clear. Do you know any example values where this does not work?

Reply:
You're very right, this rabbit hole is a lot deeper than I thought. I have tried to some llms and some googeling to help me find a satisfying solution, but to no avail. In the interest of getting this PR done and not letting Alexandre wait, do you mind just ignoring this edge case for now? It is highly unlikely that a hike passes through this line, and this edge case is left untreated in other parts of the code too, according to Alexandre. We can perhaps address it in a later sprint?

Reply:
Fair, this formula should be fine for now, I added an item in the product backlog, but I'm not even sure it is a problem to be honest.